state/state.go

```diff
@@ -257,4 +259,40 @@ type Server struct {
 
 	CollectionStatus      CollectionStatus
 	CollectionStatusMutex *sync.Mutex
+
+	// Boolean flags for which log lines should be ignored for processing
+	//
+	// Internally this uses atomics (not a mutex) due to noticable performance
+	// differences (see https://groups.google.com/g/golang-nuts/c/eIqkhXh9PLg),
+	// as we access this in high frequency log-related code paths.
+	LogIgnoreFlags uint32
+}
+
+const (
+	LOG_IGNORE_STATEMENT uint32 = 1 << iota
+	LOG_IGNORE_DURATION
+)
+
+func (s *Server) SetLogIgnoreFlags(ignoreStatement bool, ignoreDuration bool) {
+	var newFlags uint32
+	if ignoreStatement {
+		newFlags |= LOG_IGNORE_STATEMENT
+	}
```

Comment:
This will set the flags if they're not set, but the old code was explicitly written to also clear them when they are set (so that you don't need to restart the collector after adjusting settings). I think this should be something like (ignoring the concurrency issue for now):

```golang
var newFlags uint32 = s.LogIgnoreFlags
if ignoreStatement {
  newFlags |= LOG_IGNORE_STATEMENT
} else {
  newFlags &^= LOG_IGNORE_STATEMENT
}
if ignoreDuration {
  newFlags |= LOG_IGNORE_DURATION
} else {
  newFlags &^= LOG_IGNORE_DURATION
}

s.LogIgnoreFlags = newFlags
```

Reply:
The idea was to always reset the flags, i.e. `s.LogIgnoreFlags = newFlags` (or rather the atomic version of it), and thus if the passed in flags are false, there is no need to update the flags variable.

Reply:
Ah, got it. Yeah, assuming you're passing in the full state each time (which we are here), that works.